File: src/atcoder/ABC149/c/Main.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x = Integer.valueOf(sc.next());

        // エラトステネスの篩　テンプレ
        int sz = 1000000;
        // 素数リストを初期化
        int[] plist = new int[sz];
        for (int i = 0; i < plist.length; i++) {
            plist[i] = i;
        }
        // 0,1は素数ではないので除外
        plist[0]=0;
        plist[1]=0;

        // 計算対象はサイズの平方根まで
        int last = (int) Math.sqrt(sz) + 1;
        // 2以上の数字について、倍数を消していく
        for (int i = 2; i <= last; i++) {
            if (plist[i] != 0) { // plist[i]が素数でなければ
                int sz2 = (int) sz / i ;
                // リスト上の倍数の部分を0に更新していく
                for (int j = 2; j < sz2; j++) {
                    plist[i*j] = 0;
                }
            }
        }

        // 最後にできた素数リストを1個づつ見ていく
        for (int p : plist) {
            if (p >= x) {
                System.out.println(p);
                System.exit(0);
            }
        }
    }
}

File: src/atcoder/ABC149/c/main.py
x=int(input())

# xの範囲である10**5+1桁までを調べる
n=10**6
###############################
# エラトステネスの篩テンプレ

# 値を配列番号で初期化する(1,2,3,4,5...)
p=[i for i in range(n+1)]
# 1は素数ではない
p[1]=0

# 調べる数は sqrt(n)+1まで
root_n=int(n**0.5)+1
# 2以上の数字について1個づつ調べていく
for i in range(1,root_n):
    # 配列の0は調べる対象外になっているためチェックしない
    if p[i] != 0:
        # iの倍数について、0にして消していく
        for j in range(i,int(n//i+1)):
            p[i*j]=0

# リスト上の重複する0を消すためにsetにして戻す。
plist=sorted(list(set(p)))
#print(plist)
###############################

# 素数リストを順に調べていってx以上のものを調べる
for i in plist:
    if i >= x:
        print(i)
        exit()